tests/backends/hardware/sr/v4/test_power_board.py

```diff
@@ -0,0 +1,470 @@
+
+
+def test_usb_communication_error():
+    """Test that USBCommunicationError works."""
+    u = USBCommunicationError(usb.core.USBError("Test."))
+    assert str(u) == "Test."
+    assert issubclass(USBCommunicationError, Exception)
+
+
+def test_usb_error_handler_decorator():
+    """Test that the handle_usb_error decorator works."""
+    @handle_usb_error
+    def test_func():
+        raise usb.core.USBError("Test")
+
+    with pytest.raises(USBCommunicationError):
+        test_func()
+
+
+class MockBoard(Board):
+    """
+    Never called.
+
+    To be removed after refactor. See #116.
+    """
+
+    @property
+    def name(self) -> str:
+        """Name of the board."""
+        pass
+
+    @property
+    def serial(self) -> str:
+        """The serial number."""
+        pass
+
+    @property
+    def firmware_version(self) -> Optional[str]:
+        """Firmware version."""
+        pass
+
+    def make_safe(self) -> None:
+        """Make everything safe."""
+        pass
+
+    @staticmethod
+    def supported_components() -> List[Type["Component"]]:
+        """The supported components."""
+        pass
+
+    @staticmethod
+    def discover(backend: Backend) -> List["Board"]:
+        """Discover boards."""
+        pass
+
+
+class MockUSBContext:
+    """This class mocks the behaviour of usb.core.Context."""
+
+    def dispose(self, device: "MockUSBPowerBoardDevice"):
+        """Dispose of the device."""
+        pass
+
+
+class MockUSBPowerBoardDevice:
+    """This class mocks the behaviour of a USB device for a Power Board."""
+
+    def __init__(self, serial_number: str, fw_version: int = 3):
+        self.serial = serial_number
+        self.firmware_version = fw_version
+        self._ctx = MockUSBContext()
+
+    def serial_number(self) -> str:
+        """Get the serial number of the USB device."""
+        return self.serial
+
+    def ctrl_transfer(
+        self,
+        bmRequestType: int,
+        bRequest: int,
+        wValue: int = 0,
+        wIndex: int = 0,
+        data_or_wLength: Optional[Union[int, bytes]] = None,
+        timeout: Optional[int] = None,
+    ) -> bytes:
+        """Mock a control transfer."""
+        assert bRequest == 64  # This is the same for read and write.
+
+        if bmRequestType == 0x80:
+            return self.read_data(wValue, wIndex, data_or_wLength, timeout)
+        if bmRequestType == 0x00:
+            self.write_data(wValue, wIndex, data_or_wLength, timeout)
+            return b""
+
+        raise ValueError("Invalid Request Type for mock device.")
+
+    def read_data(
+        self,
+        wValue: int = 0,
+        wIndex: int = 0,
+        data_or_wLength: Optional[Union[int, bytes]] = None,
+        timeout: Optional[int] = None,
+    ) -> bytes:
+        """Mock reading data from a device."""
+        assert wValue == 0  # Always 0 on read.
+
+        if 0 <= wIndex < 6:
+            return self.read_output(data_or_wLength)
+        if wIndex == 7:
```

Comment:
I've noticed a pattern of not using `elif` for mutually exclusive code branches if each block ends in a `return` in a few of your PRs recently. Is this just personal preference or is there a PEP for this that I should be following in my own code?

Reply:
It's just my style. Happy to change if it's disliked.